packages/teleport/src/services/ssh/makeSession.ts

```diff
@@ -53,4 +54,13 @@ export function makeParticipant(json): Participant {
   };
 }
 
+/**
+ * getClusterIDFromURL solves empty clusterID result with Teleport
+ * Node's using versions < 4.3 (clusterId is not set for older versions).
+ */
+const getClusterIDFromURL = () => {
```

Comment:
We cannot reach out to the browser URL from any place in the code and manually parse parameters. Normally you access the URL from UI layer (React components + React Router). If you look at how makeSession is invoke you will see that clusterID is coming from UI components where  it already uses the clusterID to make a call to the web API. 

That said, I now think that it's better to handle this use-case on the WebAPI layer. That would be a better place for the temporary fix as this is where missing id is coming from (this is not UI bug).

Reply:
created PR to apply patch on the web apiserver instead: https://github.com/gravitational/teleport/pull/4027